test/IRGen/debug_fragment_merge.sil

```diff
@@ -3,7 +3,7 @@
 // Checking the below on 32 bit architectures would be cumbersome: each
 // fragment is 32 bits long, which changes the number of checks as well as the
 // arithmethic on the bounds of each fragment.
-// UNSUPPORTED: OS=watchos
+// UNSUPPORTED: OS=watchos, OS=linux-androideabi
```

Comment:
These two debug tests fail on 32-bit arches, as the comment says, so disable both for Android armv7 too.

Reply:
I assume that this "works" but only because for ARM64 the environment is not EABI, so you end up with the OS of `linux-android` and that would not match?

Reply:
Yes, `linux-androideabi` only matches armv7, so this test is only disabled for that 32-bit arch.

Reply:
I've removed these `debug_fragment` test changes, as more general requirements are being put in by others in #67699.